### ISSUE TITLE
[Release-1.22] - starting kubernetes: preparing server: json: cannot unmarshal string into Go value of type bootstrap.File

### ISSUE THREAD

**Issue by briandowns** [k3s-io/k3s#4732]
Backport fix for starting kubernetes: preparing server: json: cannot unmarshal string into Go value of type bootstrap.File

* #4644

**Comment by ShylajaDevadiga** [comment]
Validated the fix using **v1.22.4-rc3+k3s2 ** 
Created a two node cluster with node1 v1.21.7+k3s1 and joined node2 with v1.22.4-rc3+k3s2 

```
ubuntu@ip-172-31-8-184:~$ kubectl get nodes
NAME              STATUS   ROLES                       AGE     VERSION
ip-172-31-8-184   Ready    control-plane,etcd,master   3m4s    v1.21.7+k3s1
ip-172-31-9-55    Ready    control-plane,etcd,master   2m26s   v1.22.4-rc3+k3s2
ubuntu@ip-172-31-8-184:~$ kubectl get pods -A
NAMESPACE     NAME                                     READY   STATUS      RESTARTS   AGE
kube-system   coredns-85cb69466-bw8kq                  1/1     Running     0          2m25s
kube-system   helm-install-traefik-crd-n5c88           0/1     Completed   0          2m23s
kube-system   helm-install-traefik-kwskh               0/1     Completed   4          2m23s
kube-system   local-path-provisioner-64ffb68fd-czwtx   1/1     Running     0          2m24s
kube-system   metrics-server-9cf544f65-tls9n           1/1     Running     0          2m24s
kube-system   svclb-traefik-sxhvb                      2/2     Running     0          23s
kube-system   svclb-traefik-vpqv6                      2/2     Running     0          23s
kube-system   traefik-786ff64748-xs5vp                 1/1     Running     0          24s
```
Created a two node cluster with node1 v1.22.4+k3s1 and joined node2 with v1.22.4-rc3+k3s2 

```
ubuntu@ip-172-31-11-143:~$ kubectl get nodes
NAME               STATUS   ROLES                       AGE    VERSION
ip-172-31-11-143   Ready    control-plane,etcd,master   113s   v1.22.4+k3s1
ip-172-31-5-189    Ready    control-plane,etcd,master   6s     v1.22.4-rc3+k3s2
```

**Comment by ShylajaDevadiga** [comment]
Closing as the issue seems to be resolved as mentioned here https://github.com/k3s-io/k3s/issues/4644#issuecomment-995925219
